gammapy/modeling/fit.py

```diff
@@ -169,6 +169,10 @@ def run(self, datasets):
 
         covariance_result = self.covariance(datasets=datasets)
 
+        optimize_result.models.covariance = Covariance(
```

Comment:
I guess we can actually do the same in `Fit.run()`, such that:
```python3
result = fit.run()
result.models
``` 
Already contains the full covariance...